datafusion/physical-expr/src/aggregate/count.rs

```diff
@@ -121,14 +121,25 @@ impl GroupsAccumulator for CountGroupsAccumulator {
         // Add one to each group's counter for each non null, non
         // filtered value
         self.counts.resize(total_num_groups, 0);
-        accumulate_indices(
-            group_indices,
-            values.nulls(), // ignore values
-            opt_filter,
-            |group_index| {
-                self.counts[group_index] += 1;
-            },
-        );
+        let index_fn = |group_index| {
+            self.counts[group_index] += 1;
+        };
+
+        if values.data_type() == &DataType::Null {
+            accumulate_indices(
+                group_indices,
+                values.logical_nulls().as_ref(),
+                opt_filter,
+                index_fn,
+            );
+        } else {
+            accumulate_indices(
+                group_indices,
+                values.nulls(),
```

Comment:
I think we can use `logical_nulls` for all cases.

Reply:
`logical_null` returns an owned value, so in most cases it will clone, that's why I added a branch

Reply:
Oh ok, then we can use it I guess. I assumed that's why they don't want to change it in Arrow.